library/swift/test/EnvoyClientTests.swift

```diff
@@ -17,26 +17,163 @@ private final class MockEnvoyEngine: EnvoyEngine {
 }
 
 final class EnvoyClientTests: XCTestCase {
+  private var envoy: EnvoyClient!
+
+  override func setUp() {
+    super.setUp()
+    self.envoy = try! EnvoyClientBuilder()
+      .addEngineType(MockEnvoyEngine.self)
+      .build()
+  }
+
   override func tearDown() {
     super.tearDown()
     MockEnvoyHTTPStream.reset()
   }
 
-  func testNonStreamingExtensionSendsRequestDetailsThroughStream() throws {
-    let requestExpectation = self.expectation(description: "Sends request")
-    let dataExpectation = self.expectation(description: "Sends data")
-    let closeExpectation = self.expectation(description: "Calls close")
+  // MARK: - Streaming
+
+  func testStartingAStreamSendsHeaders() {
+    let expectation = self.expectation(description: "Sends stream headers")
+    let expectedHeaders = [
+      "key_1": ["value_a"],
+      ":method": ["POST"],
+      ":scheme": ["https"],
+      ":authority": ["www.envoyproxy.io"],
+      ":path": ["/test"],
+    ]
 
-    let expectedRequest = RequestBuilder(
-      method: .get, scheme: "https", authority: "www.envoyproxy.io", path: "/docs")
+    MockEnvoyHTTPStream.onHeaders = { headers, closeStream in
+      XCTAssertEqual(expectedHeaders, headers)
+      XCTAssertFalse(closeStream)
+      expectation.fulfill()
+    }
+
+    let request = RequestBuilder(
+      method: .post, scheme: "https", authority: "www.envoyproxy.io", path: "/test")
+      .addHeader(name: "key_1", value: "value_a")
       .build()
-    let expectedData = Data([1, 2, 3])
-    let expectedTrailers = ["foo": ["bar", "baz"]]
+    _ = self.envoy.start(request, handler: ResponseHandler())
+    self.wait(for: [expectation], timeout: 0.1)
+  }
+
+  func testSendingDataOnStreamPassesDataToTheUnderlyingStream() {
+    let expectation = self.expectation(description: "Sends stream data")
 
+    let expectedData = Data([0x0, 0x1, 0x2])
+    MockEnvoyHTTPStream.onData = { data, closeStream in
+      XCTAssertEqual(expectedData, data)
+      XCTAssertFalse(closeStream)
+      expectation.fulfill()
+    }
+
+    let request = RequestBuilder(
+      method: .post, scheme: "https", authority: "www.envoyproxy.io", path: "/test")
```

Comment:
```suggestion
      method: .post, scheme: "https", authority: "envoyproxy.io", path: "/test")
```

Reply:
?

Reply:
If anything I think this would probably be `api.envoyproxy.io`, but really could be any string for these tests. I'd prefer to leave it as is unless you feel strongly 🤷‍♂️